admin/react/assets/src/dashboard-app/pages/fonts/edit/preview/EditGooglePreviewItem.js

```diff
@@ -0,0 +1,217 @@
+								xmlns="http://www.w3.org/2000/svg"
+								data-font_weight={weight}
+								onClick={addWeight}
+							>
+								<path
+									d="M8.00078 1.30005C4.00078 1.30005 0.800781 4.50005 0.800781 8.50005C0.800781 12.5 4.00078 15.7 8.00078 15.7C12.0008 15.7 15.2008 12.5 15.2008 8.50005C15.2008 4.50005 12.0008 1.30005 8.00078 1.30005ZM8.00078 14.1C4.88078 14.1 2.40078 11.62 2.40078 8.50005C2.40078 5.38005 4.88078 2.90005 8.00078 2.90005C11.1208 2.90005 13.6008 5.38005 13.6008 8.50005C13.6008 11.62 11.1208 14.1 8.00078 14.1ZM8.80078 5.30005H7.20078V7.70005H4.80078V9.30005H7.20078V11.7H8.80078V9.30005H11.2008V7.70005H8.80078V5.30005Z"
+									fill="#007CBA"
+								/>
+							</svg>
+							<span className="ml-2" data-font_weight={weight}>{__('Add', 'custom-fonts')}</span>
+						</button>
+					}
+					{ isInGoogleState &&
+						<button className="flex text-sm text-neutral items-center py-2 px-3 border border-neutral" data-font_weight={weight} onClick={removeWeight}>
+							<svg
+								width="16"
+								height="17"
+								viewBox="0 0 16 17"
+								fill="none"
+								xmlns="http://www.w3.org/2000/svg"
+								data-font_weight={weight}
+								onClick={removeWeight}
+							>
+								<path
+									d="M8.00078 1.30005C4.00078 1.30005 0.800781 4.50005 0.800781 8.50005C0.800781 12.5 4.00078 15.7 8.00078 15.7C12.0008 15.7 15.2008 12.5 15.2008 8.50005C15.2008 4.50005 12.0008 1.30005 8.00078 1.30005ZM8.00078 14.1C4.88078 14.1 2.40078 11.62 2.40078 8.50005C2.40078 5.38005 4.88078 2.90005 8.00078 2.90005C11.1208 2.90005 13.6008 5.38005 13.6008 8.50005C13.6008 11.62 11.1208 14.1 8.00078 14.1ZM4.80078 7.70005V9.30005H11.2008V7.70005H4.80078Z"
+									fill="#7E7E7E"
+								/>
+							</svg>
+							<span className="ml-2" data-font_weight={weight}>{__('Remove', 'custom-fonts')}</span>
+						</button>
+					}
+				</div>
+			</div>
+		</div>
+	);
+};
+
+const EditGooglePreviewItem = ( { fontId, fontName } ) => {
+	const dispatch = useDispatch();
+	const editFontId = parseInt( fontId );
+
+	let editingFontData = null;
+	const [variationToggleStyle, setVariationToggleStyle] = useState('');
+
+	const restAllData = useSelector( ( state ) => state.fonts );
+
+	let toBeEditFont = {};
+	let variations = null;
+	restAllData.forEach(function(individualFont) {
+		if ( editFontId === individualFont.id ) {
+			const gFontData = bsf_custom_fonts_admin.googleFonts[individualFont.title];
+			variations = gFontData[0] ? gFontData[0] : [];
+			toBeEditFont = individualFont;
+		}
+	});
+
+	if ( undefined === toBeEditFont['fonts-data'] || ! toBeEditFont['fonts-data'].length ) {
+		editingFontData = toBeEditFont['fonts-data'];
+	}
+
+	const [editFontData, setEditGoogleFontData] = useState( editingFontData );
+	if ( null === variations ) {
+		return;
+	}
+
+	useEffect( () => {
+		let newStyle = '';
+		Object.keys( editFontData.variations ).map( ( index ) => {
+			const varwt = (editFontData.variations[index].font_weight).toString();
```

Comment:
other then varwt maybe we can rename to a batter variable name @imnavanath

Reply:
DONE